code/scripts/controllers/econsent/tasksController.js

```diff
@@ -14,9 +14,45 @@ export default class eDiaryController extends WebcController {
         this._attachHandlerPREMAndPROM();
         this.TaskService = new TaskService();
         this.model = this.getDefaultModel();
-        console.log(this.model.toObject());
+        this.renderTasks();
     }
 
+    renderTasks(){
+        const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+        for(let i = 0; i < this.model.tasks.item.length; i++){
+
+            const startDate = new Date(this.model.tasks.item[i].schedule.startDate);
+            const endDate = new Date(this.model.tasks.item[i].schedule.endDate);
+            const clickedDate = new Date(this.getState().year, months.indexOf(this.getState().month), this.getState().day);
```

Comment:
Don't abuse of getState method, you already have the state elements inside your model and avoid such long chains on processing data like `this.model.tasks.item......`
You can simply get from the model what you need to use in your appointments processing. 
For example: 
- `const {day, month, year} = this.model;`
- `const tasksItemList = this.model.toObject("tasks.item");` // this will get you all the items from the tasks sub-model and you will have smaller chains.

After you finish the data processing, you can simply assign back to the model the items.
`this.model.tasks.item = JSON.parse(JSON.stringify(tasksItemList));`